sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/spark.py

```diff
@@ -246,7 +259,11 @@ def resources(self, value: Union[Dict[str, str], SparkResourceConfiguration, Non
     def identity(
         self,
     ) -> Optional[Union[ManagedIdentityConfiguration, AmlTokenConfiguration, UserIdentityConfiguration]]:
-        """Identity that spark job will use while running on compute."""
+        """The identity that the Spark job will use while running on compute.
```

Comment:
Since the inputted identity can also be a dict, should this return type include a dict options as well? This question also applies to the setter below.

Reply:
Nope - it will never return a Dict. Look at the implementation of the setter - if the value is a dictionary, it gets transformed into one of the Configuration classes before being set.